src/sqlfluff/rules/L007.py

```diff
@@ -108,8 +121,6 @@ def _eval(self, context: RuleContext) -> Optional[List[LintResult]]:
             )
             results.append(lint_res)
 
-        if len(results) == 0:
```

Comment:
Until recently, returning an empty list didn't work correctly. Now it's okay, so remove these unnecessary lines.